gsgmail_connector.py

```diff
@@ -765,74 +746,81 @@ def _handle_test_connectivity(self, param):
         self.save_progress("Test Connectivity Passed")
         return action_result.set_status(phantom.APP_SUCCESS)
 
-    def _get_email_ids_to_process(self, service, action_result, max_results, ingest_manner,
-                                  user_id='me', labels=[], include_spam_trash=False, q=None, include_sent=False,
-                                  use_ingest_limit=False):
+    def _get_email_ids_to_process(
+        self,
+        service,
+        action_result,
+        max_results,
+        ingest_manner,
+        user_id="me",
+        labels=[],
+        include_spam_trash=False,
+        q=None,
+        include_sent=False,
+        use_ingest_limit=False,
+    ):
 
-        kwargs = {
-            'userId': user_id,
-            'includeSpamTrash': include_spam_trash,
-            'maxResults': GSMAIL_MAX_RESULT
-        }
+        kwargs = {"userId": user_id, "includeSpamTrash": include_spam_trash, "maxResults": GSMAIL_MAX_RESULT}
 
         label_ids = []
         if labels:
-            if 'labels' not in self._state:
-                self._state['labels'] = {}
+            if "labels" not in self._state:
+                self._state["labels"] = {}
             for label in labels:
-                if label.lower() not in self._state['labels']:
+                if label.lower() not in self._state["labels"]:
                     try:
                         response = service.users().labels().list(userId=user_id).execute()  # pylint: disable=E1101
-                        gmail_labels = response['labels']
+                        gmail_labels = response["labels"]
                         for gmail_label in gmail_labels:
-                            if gmail_label['name'].lower() == label.lower():
-                                self._state['labels'][label.lower()] = gmail_label['id']
+                            if gmail_label["name"].lower() == label.lower():
+                                self._state["labels"][label.lower()] = gmail_label["id"]
                     except errors.HttpError as error:
                         return action_result.set_status(phantom.APP_ERROR, error), None
-                if label.lower() not in self._state['labels']:
+                if label.lower() not in self._state["labels"]:
                     return action_result.set_status(phantom.APP_ERROR, 'Unable to find label "{}"'.format(label)), None
-                label_ids.append(self._state['labels'][label.lower()])
+                label_ids.append(self._state["labels"][label.lower()])
 
         if label_ids:
-            kwargs['labelIds'] = label_ids
+            kwargs["labelIds"] = label_ids
 
         query = []
         if q:
             query.append(q)
         if not include_sent:
-            query.append('-in:sent')
+            query.append("-in:sent")
 
         using_oldest = ingest_manner == GSMAIL_OLDEST_INGEST_MANNER
```

Comment:
Is this variable now unused? And is that on purpose?

Reply:
This variable was added when the `on poll` action was added but it has been unused since the beginning, so I believe we can remove this variable as it would not affect any existing functionality. `on poll` fetches all mail including the sent mail and will continue to do so after removing this.